data/pigui/modules/radar.lua

```diff
@@ -304,27 +390,78 @@ local function displayRadar()
 		toggle_radar = true
 	end
 
-	-- Handle mouse if it is in the radar area
-	local mp = ui.getMousePos()
-	-- TODO: adjust properly for 3D radar; bit more challenging as its an ellipse
-	if (mp - center):length() < radar2d.getRadius(radar2d) then
-		ui.popup("radarselector", function()
-			if ui.selectable(lui.HUD_2D_RADAR, shouldDisplay2DRadar, {}) then
+	--
+	-- popup functions
+	--
+	ui.popup("radarselector", function()
+		if ui.selectable(lui.HUD_2D_RADAR, shouldDisplay2DRadar, {}) then
+			if not shouldDisplay2DRadar then
 				toggle_radar = true
 			end
-			if ui.selectable(lui.HUD_3D_RADAR, not shouldDisplay2DRadar, {}) then
+			radar_popup_displayed = false
+		end
+		if ui.selectable(lui.HUD_3D_RADAR, not shouldDisplay2DRadar, {}) then
+			if shouldDisplay2DRadar then
 				toggle_radar = true
 			end
-		end)
+			radar_popup_displayed = false
+		end
+	end)
+	ui.popup("radartargetselector", function()
+		for k,v in pairs(popup_targets) do
+			if ui.selectable(v.label) then
+				onTargetClicked(v)
+				radar_popup_displayed = false
+			end
+		end
+	end)
 
-		if ui.isMouseClicked(1) then
+	--
+	-- mouse handling
+	--
+	local isMouseOverRadar = function()
+		if shouldDisplay2DRadar then
+			local mp = ui.getMousePos()
+			return (mp - center):length() < radar2d.getRadius(radar2d)
+		end
+		return ui.isMouseHoveringRect(
+			Vector2(center.x - ui.reticuleCircleRadius * 0.9,
+			        center.y - ui.reticuleCircleRadius * 0.7),
+			Vector2(center.x + ui.reticuleCircleRadius * 0.9,
+			        center.y + ui.reticuleCircleRadius * 0.7))
+	end
+
+	if radar_popup_displayed then
+		if ui.isMouseReleased(0) or ui.isMouseReleased(1) then
+			radar_popup_displayed = false
+			click_on_radar = false
+		end
+	elseif isMouseOverRadar() then
```

Comment:
Note that instead of calling `isMouseOverRadar()`, you can instead potentially submit a `ui.invisibleButton` with the appropriate button flags (might need to expose a few more in `src/lua/LuaPiGui.cpp`) before drawing the radars and test `ui.isItemClicked()` / `ui.isItemHovered()` etc.

This would allow for each radar to define its own "click area" and potentially unify some of the "clicked on radar" handling.

Reply:
Can you draw circular buttons? And how will this invisible button affect the other buttons because I don't think I'll be able to make an elliptical button!

Reply:
All interaction primitives are based on rectangles, including the window background behind the radar which is blocking mouse events from reaching the rest of the game code - so no elliptical buttons.

The overlap case definitely complicates things. Handling it the way you're currently doing it is "probably" simpler - while I could potentially add an invisible elliptical button primitive, it's simpler to just fake it at this level.

Reply:
So let's leave it for next version of ImGui and/or refactoring when/if radar equipment comes along?

Reply:
Sounds good! I mentioned this originally just to ensure you were aware of the alternate method :smile: